lib/cocoapods-core/source.rb

```diff
@@ -201,7 +201,8 @@ def search(query)
       if query.is_a?(Dependency)
         query = query.root_name
       end
-      if (specs_dir + query).directory?
+      spec_dir = specs_dir + query
+      if Pathname.glob(spec_dir) == [spec_dir]
```

Comment:
We should escape this path given to glob, since it could contain characters causing it to break. See https://github.com/CocoaPods/CocoaPods/commit/f4df7a6121e6bf4c04b44b414abe91f91b64f79a